From 2a8256ac11607ca1403b8ac0aec0cbe698694092 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Mon, 24 Jan 2022 16:40:24 +0100
Subject: [PATCH 0633/1695] plat-stm32mp1: conf: enable timers and lptimer
 drivers

Adds the STM32_TIM and STM32_LPTIMER driver support on
stm32mp1 platforms.
It will call the associated counter driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0dde406640770400d8561b6210c0b50a4dd322e8 plat-stm32mp1: conf: enable timers and lptimer drivers]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I3d25f541573d16d791646c2329c381d648df71a2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/286270
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 58746db78..aa09b13f0 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -264,11 +264,13 @@ CFG_STM32_ETZPC ?= y
 CFG_STM32_GPIO ?= y
 CFG_STM32_I2C ?= y
 CFG_STM32_IWDG ?= y
+CFG_STM32_LPTIMER ?= y
 CFG_STM32_RNG ?= y
 CFG_STM32_RSTCTRL ?= y
 CFG_STM32_RTC ?= y
 CFG_STM32_SAES ?= y
 CFG_STM32_TAMP ?= y
+CFG_STM32_TIM ?= y
 CFG_STM32_UART ?= y
 CFG_STPMIC1 ?= y
 CFG_SYSCFG ?= y
@@ -296,6 +298,10 @@ ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_SAES),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
+ifeq ($(call cfg-one-enabled, CFG_STM32_LPTIMER CFG_STM32_TIM),y)
+$(call force,CFG_COUNTER_DRIVER,y)
+endif
+
 CFG_DRIVERS_RSTCTRL ?= $(CFG_STM32_RSTCTRL)
 $(eval $(call cfg-depends-all,CFG_STM32_RSTCTRL,CFG_DRIVERS_RSTCTRL))
 
-- 
2.47.2

